### PR TITLE
Get the plugin working in UE 5.5 Preview.

### PR DESCRIPTION
**Pull request by kring**
Depends on CesiumGS/cesium-native#977.

~Right now, this most likely won't work in any _other_ UE version. But it does work (at least on Windows) in the UE 5.5 Preview that is currently available from the Epic Launcher.~ It works in all versions now.

### CODE REVIEW

**Comment by kring** [comment]
I fixed the UE 5.5 test failure. The problem was that, when running the tests headless on CI, the `Cesium3DTileset::GetCameras` was not finding any cameras. And as a result, we were never calling `Tileset::updateView`. That meant that `sampleHeightMostDetailed` requests never completed, and eventually the tests would time out. In previous versions of UE, there was a camera of some sort available, even when headless.

I fixed this by calling `updateView` even if there are no cameras. `Tileset` tolerates this well enough. I'm a little worried this could impact tile selection, though, leading to some kind of artifact in the Editor. For example, when the Editor loses focus or is minimized, and so none of the Editor viewports are valid, what happens to existing tiles? Well, I tried it, and couldn't see any problems. I'm mentioning it just in case whoever reviews this PR wants to try some additional tests in this areas.

File: Shaders/Private/CesiumPointAttenuationVertexFactory.ush [hunk 1]
```diff
@@ -357,6 +357,16 @@ FMaterialVertexParameters GetMaterialVertexParameters(
   	return Result;
 }
 
+FMaterialVertexParameters GetMaterialVertexParameters(
+  	FVertexFactoryInput Input,
+  	FVertexFactoryIntermediates Intermediates,
+  	float3 WorldPosition,
+  	half3x3 TangentToLocal,
+		bool noIdeaWhatThisIs)
```
Comment (j9liu) [inline Shaders/Private/CesiumPointAttenuationVertexFactory.ush:365]: Pulled this from other vertex factory files 😛 

```suggestion
		bool bIsPreviousFrame = false)
```

**Review by j9liu — COMMENTED** [review]
Thanks @kring ! I'm going to test this locally before I merge, but from a glance there don't appear to be any surprises.

Just a few questions / comments.

**Comment by j9liu** [comment]
**TODO** for me: Check out the point cloud vertex factory file for potential updates with the new `bool` parameter

**Comment by j9liu** [comment]
@kring I'm not sure why, but updating cesium-native to the latest commit on `main` is resulting in build errors. It looks like it's a bunch of linker errors with s2?

```
6>   Creating object C:\Dev\cesium-unreal-samples\Plugins\cesium-unreal\Binaries\Win64\UnrealEditor-CesiumRuntime.exp
6>s2.lib(s2cell_id.cc.obj) : error LNK2019: unresolved external symbol "class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > __cdecl absl::lts_20240722::str_format_internal::FormatPack(class absl::lts_20240722::str_format_internal::UntypedFormatSpecImpl,class absl::lts_20240722::Span<class absl::lts_20240722::str_format_internal::FormatArgImpl const >)" (?FormatPack@str_format_internal@lts_20240722@absl@@YA?AV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@VUntypedFormatSpecImpl@123@V?$Span@$$CBVFormatArgImpl@str_format_internal@lts_20240722@absl@@@23@@Z) referenced in function "public: void __cdecl S2Error::Init<>(enum S2Error::Code,class absl::lts_20240722::str_format_internal::FormatSpecTemplate<> const &)" (??$Init@$$V@S2Error@@QEAAXW4Code@0@AEBV?$FormatSpecTemplate@$S@str_format_internal@lts_20240722@absl@@@Z)
6>s2.lib(s2latlng.cc.obj) : error LNK2001: unresolved external symbol "class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > __cdecl absl::lts_20240722::str_format_internal::FormatPack(class absl::lts_20240722::str_format_internal::UntypedFormatSpecImpl,class absl::lts_20240722::Span<class absl::lts_20240722::str_format_internal::FormatArgImpl const >)" (?FormatPack@str_format_internal@lts_20240722@absl@@YA?AV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@VUntypedFormatSpecImpl@123@V?$Span@$$CBVFormatArgImpl@str_format_internal@lts_20240722@absl@@@23@@Z)
6>s2.lib(s2cell_id.cc.obj) : error LNK2019: unresolved external symbol "unsigned int __cdecl absl::lts_20240722::base_internal::SpinLockWait(struct std::atomic<unsigned int> *,int,struct absl::lts_20240722::base_internal::SpinLockWaitTransition const * const,enum absl::lts_20240722::base_internal::SchedulingMode)" (?SpinLockWait@base_internal@lts_20240722@absl@@YAIPEAU?$atomic@I@std@@HQEBUSpinLockWaitTransition@123@W4SchedulingMode@123@@Z) referenced in function "void __cdecl absl::lts_20240722::base_internal::CallOnceImpl<class <lambda_79ba321403081603635db02b57b3fb8e> >(struct std::atomic<unsigned int> *,enum absl::lts_20240722::base_internal::SchedulingMode,class <lambda_79ba321403081603635db02b57b3fb8e> &&)" (??$CallOnceImpl@V<lambda_79ba321403081603635db02b57b3fb8e>@@$$V@base_internal@lts_20240722@absl@@YAXPEAU?$atomic@I@std@@W4SchedulingMode@012@$$QEAV<lambda_79ba321403081603635db02b57b3fb8e>@@@Z)
6>s2.lib(s2cell_id.cc.obj) : error LNK2019: unresolved external symbol AbslInternalSpinLockWake_lts_20240722 referenced in function "void __cdecl absl::lts_20240722::base_internal::CallOnceImpl<class <lambda_79ba321403081603635db02b57b3fb8e> >(struct std::atomic<unsigned int> *,enum absl::lts_20240722::base_internal::SchedulingMode,class <lambda_79ba321403081603635db02b57b3fb8e> &&)" (??$CallOnceImpl@V<lambda_79ba321403081603635db02b57b3fb8e>@@$$V@base_internal@lts_20240722@absl@@YAXPEAU?$atomic@I@std@@W4SchedulingMode@012@$$QEAV<lambda_79ba321403081603635db02b57b3fb8e>@@@Z)
6>s2.lib(s2cell_id.cc.obj) : error LNK2019: unresolved external symbol "public: void __cdecl absl::lts_20240722::strings_internal::StringifySink::Append(class absl::lts_20240722::string_view)" (?Append@StringifySink@strings_internal@lts_20240722@absl@@QEAAXVstring_view@34@@Z) referenced in function "void __cdecl absl::lts_20240722::AbslStringify<class absl::lts_20240722::strings_internal::StringifySink>(class absl::lts_20240722::strings_internal::StringifySink &,struct absl::lts_20240722::Hex)" (??$AbslStringify@VStringifySink@strings_internal@lts_20240722@absl@@@lts_20240722@absl@@YAXAEAVStringifySink@strings_internal@01@UHex@01@@Z)
6>s2.lib(s2cell_id.cc.obj) : error LNK2019: unresolved external symbol "char * __cdecl absl::lts_20240722::numbers_internal::FastIntToBuffer(int,char *)" (?FastIntToBuffer@numbers_internal@lts_20240722@absl@@YAPEADHPEAD@Z) referenced in function "public: __cdecl absl::lts_20240722::AlphaNum::AlphaNum(int)" (??0AlphaNum@lts_20240722@absl@@QEAA@H@Z)
6>s2.lib(s2cell_id.cc.obj) : error LNK2019: unresolved external symbol "class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > __cdecl absl::lts_20240722::StrCat(class absl::lts_20240722::AlphaNum const &,class absl::lts_20240722::AlphaNum const &)" (?StrCat@lts_20240722@absl@@YA?AV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@AEBVAlphaNum@12@0@Z) referenced in function "public: class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > __cdecl S2CellId::ToString(void)const " (?ToString@S2CellId@@QEBA?AV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@XZ)
6>s2.lib(s2cell_id.cc.obj) : error LNK2001: unresolved external symbol "char const * const absl::lts_20240722::numbers_internal::kHexTable" (?kHexTable@numbers_internal@lts_20240722@absl@@3QBDB)
6>s2.lib(s2latlng.cc.obj) : error LNK2019: unresolved external symbol "public: __cdecl absl::lts_20240722::log_internal::LogMessage::OstreamView::OstreamView(struct absl::lts_20240722::log_internal::LogMessage::LogMessageData &)" (??0OstreamView@LogMessage@log_internal@lts_20240722@absl@@QEAA@AEAULogMessageData@1234@@Z) referenced in function "public: class absl::lts_20240722::log_internal::LogMessage & __cdecl absl::lts_20240722::log_internal::LogMessage::operator<<<class S2LatLng,0>(class S2LatLng const &)" (??$?6VS2LatLng@@$0A@@LogMessage@log_internal@lts_20240722@absl@@QEAAAEAV0123@AEBVS2LatLng@@@Z)
6>s2.lib(s2latlng.cc.obj) : error LNK2019: unresolved external symbol "public: virtual __cdecl absl::lts_20240722::log_internal::LogMessage::OstreamView::~OstreamView(void)" (??1OstreamView@LogMessage@log_internal@lts_20240722@absl@@UEAA@XZ) referenced in function "public: class absl::lts_20240722::log_internal::LogMessage & __cdecl absl::lts_20240722::log_internal::LogMessage::operator<<<class S2LatLng,0>(class S2LatLng const &)" (??$?6VS2LatLng@@$0A@@LogMessage@log_internal@lts_20240722@absl@@QEAAAEAV0123@AEBVS2LatLng@@@Z)
6>s2.lib(s2latlng.cc.obj) : error LNK2019: unresolved external symbol "public: class std::basic_ostream<char,struct std::char_traits<char> > & __cdecl absl::lts_20240722::log_internal::LogMessage::OstreamView::stream(void)" (?stream@OstreamView@LogMessage@log_internal@lts_20240722@absl@@QEAAAEAV?$basic_ostream@DU?$char_traits@D@std@@@std@@XZ) referenced in function "public: class absl::lts_20240722::log_internal::LogMessage & __cdecl absl::lts_20240722::log_internal::LogMessage::operator<<<class S2LatLng,0>(class S2LatLng const &)" (??$?6VS2LatLng@@$0A@@LogMessage@log_internal@lts_20240722@absl@@QEAAAEAV0123@AEBVS2LatLng@@@Z)
6>s2.lib(s2latlng.cc.obj) : error LNK2019: unresolved external symbol "private: void __cdecl absl::lts_20240722::log_internal::LogMessage::CopyToEncodedBuffer<0>(class absl::lts_20240722::string_view)" (??$CopyToEncodedBuffer@$0A@@LogMessage@log_internal@lts_20240722@absl@@AEAAXVstring_view@23@@Z) referenced in function "public: class absl::lts_20240722::log_internal::LogMessage & __cdecl absl::lts_20240722::log_internal::LogMessage::operator<<<34>(char const (&)[34])" (??$?6$0CC@@LogMessage@log_internal@lts_20240722@absl@@QEAAAEAV0123@AEAY0CC@$$CBD@Z)
6>s2.lib(s2latlng.cc.obj) : error LNK2019: unresolved external symbol "private: static bool __cdecl absl::lts_20240722::str_format_internal::FormatArgImpl::Dispatch<double>(union absl::lts_20240722::str_format_internal::FormatArgImpl::Data,class absl::lts_20240722::str_format_internal::FormatConversionSpecImpl,void *)" (??$Dispatch@N@FormatArgImpl@str_format_internal@lts_20240722@absl@@CA_NTData@0123@VFormatConversionSpecImpl@123@PEAX@Z) referenced in function "public: __cdecl absl::lts_20240722::str_format_internal::FormatArgImpl::FormatArgImpl<double>(double const &)" (??$?0N@FormatArgImpl@str_format_internal@lts_20240722@absl@@QEAA@AEBN@Z)
6>s2.lib(coder.cc.obj) : error LNK2019: unresolved external symbol "public: __cdecl absl::lts_20240722::log_internal::LogMessageFatal::LogMessageFatal(char const *,int,class absl::lts_20240722::string_view)" (??0LogMessageFatal@log_internal@lts_20240722@absl@@QEAA@PEBDHVstring_view@23@@Z) referenced in function "public: __cdecl Encoder::~Encoder(void)" (??1Encoder@@QEAA@XZ)
6>s2.lib(coder.cc.obj) : error LNK2019: unresolved external symbol "public: __cdecl absl::lts_20240722::log_internal::LogMessageFatal::~LogMessageFatal(void)" (??1LogMessageFatal@log_internal@lts_20240722@absl@@QEAA@XZ) referenced in function "public: __cdecl Encoder::~Encoder(void)" (??1Encoder@@QEAA@XZ)
6>s2.lib(coder.cc.obj) : error LNK2019: unresolved external symbol "class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > * __cdecl absl::lts_20240722::log_internal::MakeCheckOpString<unsigned char const *,unsigned char const *>(unsigned char const *,unsigned char const *,char const *)" (??$MakeCheckOpString@PEBEPEBE@log_internal@lts_20240722@absl@@YAPEAV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@PEBE0PEBD@Z) referenced in function "class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > * __cdecl absl::lts_20240722::log_internal::Check_LEImpl<unsigned char *,unsigned char *>(unsigned char * const &,unsigned char * const &,char const *)" (??$Check_LEImpl@PEAEPEAE@log_internal@lts_20240722@absl@@YAPEAV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@AEBQEAE0PEBD@Z)
6>C:\Dev\cesium-unreal-samples\Plugins\cesium-unreal\Binaries\Win64\UnrealEditor-CesiumRuntime.dll : fatal error LNK1120: 15 unresolved externals
```
I made sure to delete the existing `build` folder in `extern`, as well as `Source/ThirdParty`. But seems like this is happening consistently

**Comment by lilleyse** [comment]
I'm getting the same linker errors when building cesium-native by itself with gcc 13

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -384,3 +384,121 @@ jobs:
       test-package-base-name: "CesiumForUnreal-54"
       visual-studio-version: "2022"
       visual-studio-components: "Microsoft.VisualStudio.Component.VC.14.34.17.4.x86.x64,Microsoft.VisualStudio.Component.Windows10SDK.18362"
+  Windows55:
```
[the remark this answers is not in the document]
Comment (kring) [reply .github/workflows/build.yml:387]: Yes!

File: extern/vcpkg-overlays/abseil/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,72 @@
+if(NOT VCPKG_TARGET_IS_WINDOWS)
```
[the remark this answers is not in the document]
Comment (kring) [reply extern/vcpkg-overlays/abseil/portfile.cmake:1]: Yes, the Abseil overlay port is required to work around an Abseil bug with old-ish Android compilers.

**Comment by kring** [comment]
I hate to say it, but those linker errors are probably caused by vcpkg not picking up the Abseil overlay port. It doesn't seem to be smart enough to do that automatically when the overlay port is added. Running this and then cmake configure / build should fix it:

```
c:\.ezvcpkg\2024.11.16\vcpkg remove abseil:x64-windows-unreal --recurse
```

**Comment by j9liu** [comment]
Sorry to bother @kring, but even with that `remove` command for `vcpkg` I'm still running into the same errors. Happening not just on UE 5.5, but for 5.3 and 5.4 as well. I even went as far as to delete the `.ezvcpkg` folder entirely, to no avail.

Is there anything additional that I have to do when doing cmake configure / build? Maybe I'm missing something? 😞

**Comment by kring** [comment]
Oh it's also worth a try deleting your `Plugins\cesium-unreal\Source\ThirdParty` directory. Perhaps there's an old binary hanging around there from an install in a previous branch?

**Comment by kring** [comment]
I keep thinking of other things to try... UBT only re-runs the .build.cs files when it thinks it needs to, and re-running ours is the only way for new cesium-native .libs to be picked up. So sometimes it helps to touch (open and re-save) `CesiumRuntime.build.cs` and `CesiumEditor.build.cs` to ensure UBT re-runs them.

**Comment by kring** [comment]
@ahmeds5-90 that issue is covered here:
https://community.cesium.com/t/preview-release-available-for-unreal-engine-5-5/36544/9

**Pull request merged**